src/main/java/TaskList.java

```diff
@@ -0,0 +1,122 @@
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Class that encapsulate, control, and evaluate a list of task.
+ */
+public class TaskList {
+    private final List<Task> tasks;
+
+    /**
+     * Constructor of TaskList object
+     */
+    TaskList() {
+        this.tasks = new ArrayList<Task>();
+    }
+
+    /**
+     * Constructor of TaskList object
+     * @param tasks List of Task
+     */
+    TaskList(List<Task> tasks) {
+        List<Task> newTasks = new ArrayList<Task>();
+        for (Task task : tasks) {
+            newTasks.add(task);
+        }
+        this.tasks = newTasks;
+    }
+
+    /**
+     * Iterate Task object in list and ask Ui to print them
+     */
+    public String iterateList() {
+        if (this.tasks.size() == 0) {
+            return Ui.showMessage("There are no task in your list");
+        }
+        String output = Ui.showMessage("Here are the tasks in your list:");
+        for (int i = 0; i < this.tasks.size(); i++) {
+            output += Ui.showMessage(String.valueOf(i + 1) + "." + this.tasks.get(i));
+        }
+        return output;
+    }
+
+    /**
+     * Set a task to be done
+     * @param input The String representation of the index of the task
+     * @throws DukeException Index out of bound
+     */
+    public String finishATask(String input) throws DukeException {
```

Comment:
This function is named "finishATask" while another one is "addTask", maybe you could pick either way to name the function and standardize it across your functions so as to keep it more consistent (i.e. either name it (verb)ATask or (verb)Task)?